File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> m;
        for(int it=0;it<nums.size();it++)
        {
            int num=nums[it]; int com=target-num;
            auto i=m.find(com);
            if(i!=m.end())
            {
                return {i->second,it};
            }
            m[num]=it;
        }
        return {};
    }
};

File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hm=new HashMap();
        int[] arr=new int[2]; int pos=0;
        for(int i=0;i<nums.length;i++){
            if(hm.containsKey(target-nums[i])){
                arr[pos++]=hm.get(target-nums[i]);
                arr[pos++]=i;
                return arr;
            }
            else hm.put(nums[i],i);
        }
        return arr;
    }
}